server/functions/framework/compiled_function.go

```diff
@@ -116,8 +116,9 @@ func newCompiledFunctionInternal(
 	if returnType.IsPolymorphicType() {
 		if hasPolymorphicParam {
 			c.callResolved[len(c.callResolved)-1] = c.resolvePolymorphicReturnType(functionParameterTypes, originalTypes, returnType)
-		} else if c.Name == "array_in" || c.Name == "array_recv" {
+		} else if c.Name == "array_in" || c.Name == "array_recv" || c.Name == "enum_in" || c.Name == "enum_recv" || c.Name == "anyenum_in" || c.Name == "anyenum_recv" {
```

Comment:
Possible that the `oid` type has a special property with polymorphic return types, in that perhaps their value will set the return type in the absence of another polymorphic type in the parameter list. We should put this as a possibility in the `TODO`.